src/mark.ts

```diff
@@ -668,6 +672,43 @@ export const defaultBarConfig: RectConfig<SignalRef> = {
   binSpacing: 1,
 };
 
+export interface TextConfig<ES extends ExprRef | SignalRef> extends MarkConfig<ES> {
+  /**
+   * The type of the background of text marks: `rect` or `outline`
+   *
+   * __Default value:__  `rect`
+   */
+  bgType?: 'rect' | 'outline';
+  /**
+   * The color of the background of text marks
+   *
+   * __Default value:__  `white`
+   */
+  bgColor?: Color | ES;
+
+  /**
+   * The opacity of the background of text marks
+   *
+   * __Default value:__  `1`
+   */
+  bgOpacity?: number | ES;
+
+  /**
+   * The padding of background rectangles or the stroke-width of the outline of text marks.
+   * If an object, the value should have the format {"left": 5, "top": 5, "right": 5, "bottom": 5}.
+   *
+   * __Default value:__  `2`
+   */
+  bgPadding?: Padding;
```

Comment:
I think using padding for both rect padding and text strokeWidth is mixing up two different things a bit.


I think it's better to separate background from outline since they can have different sets of properties. 

```ts
background: Color | ES | {
   color: ... // required
   opacity: ...
   padding: 
},
outline: Color | ES | {
   color: ...
   // no padding, cornerRadius
   strokeWidth  
}
```

Reply:
If we gonna include outline in this PR, we should also include example specs for outline too.

Reply:
@kanitw If you have some time to spare, can you have a look at these changes and let me know if further work is needed? Thanks in advance.